src/odemis/dataio/catmaid.py

```diff
@@ -0,0 +1,343 @@
+        tile_shape = (stack_info["mirrors"][0]["tile_width"], stack_info["mirrors"][0]["tile_height"])
+        # resolution is given in nanometers by catmaid.
+        resolution = {k: v * 1e-9 for k, v in stack_info["resolution"].items()}
+        metadata = dict()
+        metadata[model.MD_PIXEL_SIZE] = (resolution["x"], resolution["y"])
+        dtype = numpy.uint8
+        DataArrayShadow.__init__(self, shape, dtype, metadata, maxzoom=8, tile_shape=tile_shape)
+
+        self._base_url = base_url
+        self._session = requests.Session()
+        self._stack_info = stack_info
+        file_extension = self._stack_info["mirrors"][0]["file_extension"]
+        self._file_extension = file_extension[1:] if file_extension.startswith(".") else file_extension
+
+    def getTile(self, x, y, zoom, depth=437):
+        """
+        Fetches one tile
+        x (0<=int): X index of the tile.
+        y (0<=int): Y index of the tile
+        zoom (0<=int): zoom level to use. The total shape of the image is shape / 2**zoom.
+            The number of tiles available in an image is ceil((shape//zoom)/tile_shape)
+        depth (0<=int): The Z index of the stack.
+        return:
+            tile (DataArray): tile containing the image data and the relevant metadata.
+        """
+        tile_width, tile_height = self.tile_shape
+        tile_url = format_tile_url(
+            tile_source_type=self._stack_info["mirrors"][0]["tile_source_type"],
+            image_base=self._stack_info["mirrors"][0]["image_base"],
+            zoom=zoom,
+            depth=depth,
+            col=x,
+            row=y,
+            file_extension=self._file_extension,
+            tile_width=tile_width,
+            tile_height=tile_height,
+        )
+        _, username, password = read_config_file(self._base_url, username=True, password=True)
+        try:
+            image = response_to_array(self._session.get(tile_url, auth=(username, password)))
+        except HTTPError as e:
+            if e.response.status_code == 401:
+                raise AuthenticationError("Authentication failed while getting tiles at {}".format(tile_url))
+            else:
+                logging.error("No tile at %s (error %s), returning blank tile", tile_url, e.response.status_code)
+                image = numpy.zeros((tile_width, tile_height))
+
+        tile = model.DataArray(image, self.metadata.copy())
+        orig_pixel_size = self.metadata.get(model.MD_PIXEL_SIZE, (1e-6, 1e-6))
+        # calculate the pixel size of the tile for the zoom level
+        tile_pixel_size = tuple(ps * 2 ** zoom for ps in orig_pixel_size)
+        tile.metadata[model.MD_PIXEL_SIZE] = tile_pixel_size
+
+        # calculate the center of the tile
+        position = get_tile_md_pos((x, y), self.tile_shape, tile, self)
+        tile.metadata[model.MD_POS] = [p for p in position]
+
+        return tile
+
+    def getData(self):
+        """Abstract method of DataArrayShadow"""
+        pass
```

Comment:
The idea of this method is to provide a simple way, for simple scripts to get just the entire data. I think we could have a generic function which computes that big image by just calling `.getTile()` and setting the right metadata. If it's too much work (ie, more than just moving some code from dataio.tiff), I'd suggest to `raise NotImplementedError()`.

Reply:
Did not manage to implement this quickly, so I'm raising a `NotImplementedError()`. I think if at some point we do decide to implement this, we have to keep in mind that at zoom level 0 this can be a lot of data. The projects at the public servers have over 100 by 100 tiles each 1024x1024 pixels at zoom level 0.